### PR TITLE
Blog: Hawkular and Grafana out of the box

### PR DESCRIPTION
**Pull request by jotak**

### CODE REVIEW

File: src/main/jbake/content/blog/2017/05/31/hawkular-grafana-ootb.adoc [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+:jbake-status: published
+:jbake-tags: blog, metrics, grafana, docker, vertx
+:figure-caption!:
+
+link:https://github.com/jotak/hawkular-grafana-outofthebox[Hawkular and Grafana Out of the Box] is a new way to quickly setup Hawkular and Grafana together with very little work.
+
+Assuming you already have docker installed and recent enough (you need at least Compose 1.6.0 and Docker Engine 1.10.0),
+you're just 2 lines of YAML edition and a `docker-compose up` away from having Hawkular + Grafana running and pre-configured.
+
+It is NOT intended for production/long-term usage unless you do some extra work to make it secured.
```
Comment (stefannegrea) [inline src/main/jbake/content/blog/2017/05/31/hawkular-grafana-ootb.adoc]: I would not add this line; this is not related to the technical details presented here.

File: src/main/jbake/content/blog/2017/05/31/hawkular-grafana-ootb.adoc [hunk 2]
```diff
@@ -0,0 +1,253 @@
+= Hawkular and Grafana Out of the Box
+Joel Takvorian
+2017-05-31
+:jbake-type: post
+:jbake-status: published
+:jbake-tags: blog, metrics, grafana, docker, vertx
+:figure-caption!:
+
+link:https://github.com/jotak/hawkular-grafana-outofthebox[Hawkular and Grafana Out of the Box] is a new way to quickly setup Hawkular and Grafana together with very little work.
```
Comment (stefannegrea) [inline src/main/jbake/content/blog/2017/05/31/hawkular-grafana-ootb.adoc:9]: Not sure why this text is rendered larger, but it looks weird.

File: src/main/jbake/content/blog/2017/05/31/hawkular-grafana-ootb.adoc [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+<2> Using the official Hawkular Services image. Accessible on http://localhost:8080/hawkular/metrics or http://localhost:8080/hawkular/alerts, with credentials _jdoe_/_password_.
+<3> The volume is commented-out, so data is not persisted if you run `docker-compose down`. To persist data, uncomment these lines and eventually set something different than `/tmp/opt/hawkular/server` as the host directory. Make sure this directory exists on your host.
+<4> The Hawkular agent is disabled by default, which means Hawkular is not self-monitored. You can enable it if you want, and use the `hawkular-self` dashboard.
+<5> The Cassandra node. Note that it's not exposed outside of the compose's subnetwork.
+<6> The Grafana server with pre-installed Hawkular plugin, accessible on http://localhost:3000, with credentials _admin_/_admin_
+<7> This image contains a bootstrap script that configures a Grafana datasource with the given tenant (<8>), which is also injected in dashboard templates defined here (<9>). Those dashboards are sent to Grafana through its REST API. Current available dashboard templates are listed below.
```
Comment (stefannegrea) [inline src/main/jbake/content/blog/2017/05/31/hawkular-grafana-ootb.adoc]: What is <8> and <9> ?

File: src/main/jbake/content/blog/2017/05/31/hawkular-grafana-ootb.adoc [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+
+If you want to have more than 1 tenant configured, you can add other sections to this compose file (_bootstrap2_, _bootstrap3_, etc. the name doesn't matter),
```
Comment (stefannegrea) [inline src/main/jbake/content/blog/2017/05/31/hawkular-grafana-ootb.adoc]: Please replace 1 with one.

File: src/main/jbake/content/blog/2017/05/31/hawkular-grafana-ootb.adoc [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+
+== CPU Demo
+
+Want a demo? Take the docker-compose file located in `cpudemo/` (link:https://github.com/jotak/hawkular-grafana-outofthebox/blob/master/cpudemo/docker-compose.yml[yes, this one]) and store it locally. From its location run:
+
+[source,bash]
+----
+docker-compose up
+----
+You may have to wait several minutes the first time you run it so that the docker engine downloads everything.
```
Comment (stefannegrea) [inline src/main/jbake/content/blog/2017/05/31/hawkular-grafana-ootb.adoc]: In this paragraph you use passive voice in the second sentence but the first. I would prefer the passive voice in the firs sentence too.

File: src/main/jbake/content/blog/2017/05/31/hawkular-grafana-ootb.adoc (hunk 2 above)
Comment (pilhuhn) [reply src/main/jbake/content/blog/2017/05/31/hawkular-grafana-ootb.adoc:9]: It renders larger, because it is the first paragraph, which is sort of the abstract. That is default with AsciiDoc

File: src/main/jbake/content/blog/2017/05/31/hawkular-grafana-ootb.adoc (hunk 1 above)
Comment (pilhuhn) [reply src/main/jbake/content/blog/2017/05/31/hawkular-grafana-ootb.adoc]: I don't think it is bad to mention this. Should be in a `NOTE:` section though

File: src/main/jbake/content/blog/2017/05/31/hawkular-grafana-ootb.adoc (hunk 3 above)
Comment (pilhuhn) [reply src/main/jbake/content/blog/2017/05/31/hawkular-grafana-ootb.adoc]: Those are TENANT and DASHBOARDS in the env section.
@jotak: I think they may only render correctly when you put the `<x>` on the start of a line.

File: src/main/jbake/content/blog/2017/05/31/hawkular-grafana-ootb.adoc [hunk 6]
```diff
@@ -0,0 +1,253 @@
...
+
+If you want to have more than 1 tenant configured, you can add other sections to this compose file (_bootstrap2_, _bootstrap3_, etc. the name doesn't matter),
+with the same docker image and different environment variables. You are limited, however, in the number of tenants per dashboard (a single dashboard cannot use several tenants).
+You need to duplicate dashboards, or create them manually.
+
+== Available dashboard templates
+
+Only a few are available right now, but the list should grow and pull requests are welcome!
+
+Some dashboards may be very specific, dedicated to the demos presented here, while others aim to be generic and reusable. I guess the next step will be to inject custom dashboards through volume in docker-compose without having the rebuild the bootstrap image. Anyway, the current list is:
```
Comment (pilhuhn) [inline src/main/jbake/content/blog/2017/05/31/hawkular-grafana-ootb.adoc]: s/I guess//
Another option could be to load them from a dedicated repo on github

**Comment by pilhuhn** [comment]
Awesome, great job @jotak !

File: src/main/jbake/content/blog/2017/05/31/hawkular-grafana-ootb.adoc (hunk 6 above)
Comment (jotak) [reply src/main/jbake/content/blog/2017/05/31/hawkular-grafana-ootb.adoc]: Possible, but it would still require us/me to validate the dashboards (merge PR), whereas a volume allows people to put whatever they want without having to share (think about dashboards very specific to an application, it doesn't have to be shared)

**Comment by theute** [comment]
All the 3 demos ran fine and easy to set up, that's awesome stuff :)

2 details:
   - On the first image, we see "CPU Demo" on the screenshot and the text above led me to think I would see the same (but in your case you already accessed that dashboard), but that's not where the user will access the dashboard, you'll have to click on "Home". Straightforward for anyone used to Grafana, but I had a second where I was wondering if my setup was correct (and started to think, I would have to debug things again ;))
   - In the Hawkular example, it can take several minutes for data to show up in all the sections. You could warn the user.

Those are details, if you want I can merge as-is or wait on your changes. Just let me know.

**Comment by jotak** [comment]
Thanks for the feedback, I will amend the post before you can merge

File: src/main/jbake/content/blog/2017/05/31/hawkular-grafana-ootb.adoc [hunk 7]
```diff
@@ -0,0 +1,251 @@
...
+2017-05-31
+:jbake-type: post
+:jbake-status: published
+:jbake-tags: blog, metrics, grafana, docker, vertx
+:figure-caption!:
+
+link:https://github.com/jotak/hawkular-grafana-outofthebox[Hawkular and Grafana Out of the Box] is a new way to quickly setup Hawkular and Grafana together with very little work.
+
+Assuming you already have docker installed and recent enough (you need at least Compose 1.6.0 and Docker Engine 1.10.0),
+you're just 2 lines of YAML edition and a `docker-compose up` away from having Hawkular + Grafana running and pre-configured.
```
Comment (jshaughn) [inline src/main/jbake/content/blog/2017/05/31/hawkular-grafana-ootb.adoc:12]: edition -> addition (or just omit 'edition' completely)

File: src/main/jbake/content/blog/2017/05/31/hawkular-grafana-ootb.adoc [hunk 8]
```diff
@@ -0,0 +1,251 @@
...
+      - ADMIN_TOKEN
+      - HAWKULAR_AGENT_ENABLE=false # <4>
+  hawkular-cassandra: # <5>
+    image: cassandra:3.0.9
+    environment:
+      - CASSANDRA_START_RPC=true
+  grafana: # <6>
+    image: "hawkular/hawkular-grafana-datasource:latest"
+    ports:
+      - "3000:3000"
```
Comment (jshaughn) [inline src/main/jbake/content/blog/2017/05/31/hawkular-grafana-ootb.adoc:44]: The same port as ManageIQ?  Is that what you want?

**Review by jshaughn — COMMENTED** [review]
Good stuff!

File: src/main/jbake/content/blog/2017/05/31/hawkular-grafana-ootb.adoc (hunk 7 above)
Comment (jotak) [reply src/main/jbake/content/blog/2017/05/31/hawkular-grafana-ootb.adoc:12]: 'Edition' because it requires editing an existing template. Is it correct then?

File: src/main/jbake/content/blog/2017/05/31/hawkular-grafana-ootb.adoc (hunk 8 above)
Comment (jotak) [reply src/main/jbake/content/blog/2017/05/31/hawkular-grafana-ootb.adoc:44]: lol, I didn't know, it's Grafana's default, which is I think the default for many node.js servers.

Comment (jotak) [reply src/main/jbake/content/blog/2017/05/31/hawkular-grafana-ootb.adoc:44]: You think it's important to change it?

**Comment by jotak** [comment]
@theute it should be good now

File: src/main/jbake/content/blog/2017/05/31/hawkular-grafana-ootb.adoc (hunk 7 above)
Comment (jotak) [reply src/main/jbake/content/blog/2017/05/31/hawkular-grafana-ootb.adoc:12]: The 2 lines in question are:

```yaml
      - TENANT=put_your_tenant_here
      - DASHBOARDS=put_dashboard_file_names_here # only dashboards added from Dockerfile can be set here, see README
```

File: src/main/jbake/content/blog/2017/05/31/hawkular-grafana-ootb.adoc (hunk 8 above)
Comment (jshaughn) [reply src/main/jbake/content/blog/2017/05/31/hawkular-grafana-ootb.adoc:44]: No, I was just pointing it out as a potential but unlikely conflict.

**Comment by theute** [comment]
merged !
Thanks Joel, it was great fun to give it a try !

File: src/main/jbake/content/blog/2017/05/31/hawkular-grafana-ootb.adoc (hunk 7 above)
Comment (jshaughn) [reply src/main/jbake/content/blog/2017/05/31/hawkular-grafana-ootb.adoc:12]: I don't know, I just found "edition" confusing (it's a noun) but I doubt a reader will be thrown off, the intention is still clear.  Whatever you want is fine.

**Comment by jotak** [comment]
So what's your score at "falco the hawk"? :-)
